ratel/src/astgen/expression.rs

```diff
@@ -239,7 +255,7 @@ impl<'ast> SerializeInLoc for Expression<'ast> {
         use self::Expression::*;
 
         match *self {
-            Void => unimplemented!(),
+            Void => unreachable!(),
```

Comment:
I think we should return an `Err` here. It might not be possible to get a `Void` expression from an AST produced by the parser, but it's still possible to manually construct such an AST.

That said, it might be worth rethinking having `Void` as a variant for `Expression` if the only place it's used in is arrays (and function signatures, calls?).